modin/extension/extensions.py

```diff
@@ -0,0 +1,150 @@
+from typing import Any, Union
+
+import modin.pandas
+from modin.pandas import _PD_EXTENSIONS_
+from modin.pandas.dataframe import _DATAFRAME_EXTENSIONS_, DataFrame
+from modin.pandas.series import _SERIES_EXTENSIONS_, Series
+
+
+def _set_attribute_on_obj(
+    name: str, extensions_dict: dict, obj: Union[DataFrame, Series, modin.pandas]
+):
+    """
+    Create a new or override existing attribute on obj
+
+    Parameters
+    ----------
+    name : str
+        The name of the attribute to assign to `obj`.
+    extensions_dict : dict
+        The dictionary mapping extension name to `new_attr` (assigned below).
+    obj : DataFrame, Series, or modin.pandas
+        The object we are assigning the new attribute to.
+
+    Returns
+    -------
+    decorator
+        Returns the decorator function.
+    """
+
+    def decorator(new_attr: Any):
```

Comment:
Should we use the logic with @functools.wraps(func) here?

Reply:
I think it is correct to not use `functools.wraps` here. The user is not calling the function that is being decorated, they are calling the APIs that we are registering on the modin objects, and we only want this decorator to run one time, when the functionality is registered. Here is a screenshot of how it looks without `functools.wraps`.

<img width="985" alt="Screenshot 2024-02-23 at 9 41 42 AM" src="https://github.com/modin-project/modin/assets/149107610/49dd6fe9-1ba1-4e6f-a2f7-2fbcdf5cc890">

Reply:
I couldn't test the case with `functools.wraps` to see how it would look like. I am getting this error when importing `register_dataframe_accessor`.

```python
In [1]: from modin.extension import register_dataframe_accessor

TypeError: Union[arg, ...]: each arg must be a type. Got <module 'modin.pandas' from '/home/me/repo/modin/modin/pandas/__init__.py'>.
```

Did you see a similar error?

Reply:
I don't see that error, and it passes for me and in CI. Maybe I can just change to module type from `modin.pandas`